plugins/woocommerce-admin/client/marketplace/components/content/content.tsx

```diff
@@ -40,169 +42,157 @@ export default function Content(): JSX.Element {
 	} = marketplaceContextValue;
 	const query = useQuery();
 
-	// On initial load of the in-app marketplace, fetch extensions, themes and business services
-	// and check if there are any business services available on WCCOM
+	// Function to tag products with their type
+	const tagProductsWithType = (
+		products: Product[],
+		type: ProductType
+	): Product[] => {
+		return products.map( ( product ) => ( {
+			...product,
+			type, // Adding the product type to each product
+		} ) );
+	};
+
+	// Fetch all categories when query.term or query.category changes
 	useEffect( () => {
-		const categories: Array< keyof SearchResultsCountType > = [
-			'extensions',
-			'themes',
-			'business-services',
+		const categories: Array< {
+			key: keyof SearchResultsCountType;
+			type: ProductType;
+		} > = [
+			{ key: 'extensions', type: ProductType.extension },
+			{ key: 'themes', type: ProductType.theme },
+			{ key: 'business-services', type: ProductType.businessService },
 		];
```

Comment:
This can probably go in some more generic file. We're making this connection in several places.